lms/djangoapps/grades/tests/test_tasks.py

```diff
@@ -407,17 +407,6 @@ def test_behavior(self, batch_size):
         )
 
     @ddt.data(*xrange(1, 12, 3))
-    def test_database_calls(self, batch_size):
```

Comment:
Because these kinds of tests are extremely brittle and difficult to maintain.  Any change that affects the number of calls to the database, but not the functionality, will cause them to break, which shouldn't be the case for a unit test.  I could _maybe_ see the value of a test that asserts that queries to the database are made at all, and that they don't exceed some threshold.  But having to reason about all of the magic numbers in these two lines:
```python
per_user_queries = 15 * min(batch_size, 6)  # No more than 6 due to offset
with self.assertNumQueries(6 + per_user_queries):
```
to make code changes that don't change behavior is a ton of overhead for little to no benefit.

IMHO.